framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -442,7 +443,8 @@ private Range calculateRangeBinaryOp(
                     resultRange = leftRange.bitwiseXor(rightRange);
                     break;
                 default:
-                    throw new UnsupportedOperationException();
+                    ErrorReporter.errorAbort("ValueTransfer: unsupported operation: " + op);
+                    resultRange = null;
```

Comment:
Is this code dead?  If so, please throw an exception with message "this can't happen".  Using an assignment will lead a reader to think that this statement can get executed.

Same comment elsewhere.